ros2bag_tools/ros2bag_tools/filter/cut.py

```diff
@@ -124,15 +149,30 @@ def output_size_factor(self, metadata):
             end = filter_end
         return min(1, max(0, (end - start) / metadata.duration))
 
-    def filter_msg(self, msg):
-        (_, _, t) = msg
+    def filter_topic(self, topic_metadata):
+        self._deserializer.add_topic(topic_metadata)
```

Comment:
Please initialize the deserializer only for topics of the relevant durability. Some filters, like cut, are applicable to generic messages, and don't require the underlying message type to be installed or sourced. When we try to load the type of all topics we loose that advantage.

Reply:
I added the requested change. 

Besides, cutting bags with messages that are not installed wasn't possible before my commit as well. I tried it with commit 9a0e15 because I was curious about the autoware_auto_perception_msgs warning:

```
albers@leaf-pc:/mnt/data$ ros2 bag cut --in-storage mcap --out-storage mcap --start 1 --duration 2 ./work/rosbag2_2023_01_06-15_10_54_a45
no .msg definition for autoware_auto_perception_msgs/msg/TrackedObjects, falling back to IDL
Traceback (most recent call last):
  File "/opt/ros/humble/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.18.5', 'console_scripts', 'ros2')())
  File "/opt/ros/humble/lib/python3.10/site-packages/ros2cli/cli.py", line 89, in main
    rc = extension.main(parser=parser, args=args)
  File "/opt/ros/humble/lib/python3.10/site-packages/ros2bag/command/bag.py", line 38, in main
    return extension.main(args=args)
  File "/home/albers/autoware/build/ros2bag_tools/ros2bag_tools/verb/__init__.py", line 128, in main
    writer.create_topic(topic)

```

Reply:
Need to check tomorrow with sqlite, but maybe mcap specific? mcap is supposed to be self-contained and should contain the full message definition when creating the bag.

Thx for the pr.

Reply:
You are right. The error does occur with sqlite3. It makes sense when the message definition needs to be included.

Is there anything else that needs to be done for this PR?

Reply:
I think not. Looks good. Any other changes will be integrated if the use case arises. I was not affected so far and also was not affected by the restamp case, so I see it solved.

Reply:
I agree and will merge now :+1: